src/main/java/io/jenkins/plugins/coverage/targets/CoverageResult.java

```diff
@@ -92,6 +87,8 @@
 
     private final Map<CoverageElement, Ratio> localResults = new TreeMap<>();
 
+    private Map<CoverageElement, Float> deltaResults = new TreeMap<>();
```

Comment:
Shouldn't the values be of type `Ratio` again? (Maybe we need to add a `subtract` method in `Ratio`)

Reply:
I am not sure if this is needed. The line coverage diff was saved as float until now as well. Also I am not sure how to calculate the diff between two Ratios with numerator and denominator. What exactly do they express? The diff between the two builds was calculated with the `getPercentageFloat()` method, so the coverage is calculated as `(100 * numerator / denominator)`. So if we want to save the diff as Ratio too, we basically need to subtract the numerator of Ratio1 from Ratio2 and the denominator from Ratio1 from Ratio2 and divide it, or am I forgetting something?

Reply:
I just thought since we already have that class that handles rounding, formatting, etc. why not reuse it for the delta. And to subtract two rational numbers we need both numbers use the common denominator and subtract the numerators afterwards. But if we do not need the details of `Ratio` we simply can use float values, that is fine. (When creating a delta we would also need to make `Ratio` sign aware which it is currently not).

Reply:
That might be the nicer solution. I will check it out and try to use the Ratio.

Reply:
It doesn't really make sense to me... Have now left it like this and implemented it with floats.